modules/simple-payments/paypal-express-checkout.js

```diff
@@ -119,37 +142,35 @@ var PaypalExpressCheckout = {
 					env: env
 				};
 
-				return paypal
-					.request
-					.post( PaypalExpressCheckout.getCreatePaymentEndpoint( blogId ), payload )
-					.then( function( paymentResponse ) {
-						return paymentResponse.id;
-					} )
-					.catch( function( paymentError) {
-						PaypalExpressCheckout.showError( 'Item temporarily unavailable', buttonDomId );
-					} );
+				return new paypal.Promise( function( resolve, reject ) {
+					jQuery.post( PaypalExpressCheckout.getCreatePaymentEndpoint( blogId ), payload )
+						.done( function( paymentResponse ) {
+							resolve( paymentResponse.id );
+						} )
+						.fail( function( paymentError ) {
+							var errorMessage = PaypalExpressCheckout.processErrorMessage( paymentError );
+							PaypalExpressCheckout.showError( errorMessage, buttonDomId );
+							reject( new Error( paymentError.responseJSON.code ) );
```

Comment:
I would not throw a new JS error. If we get some error on backend, we should inform the user by a front-end message, not a JS error I think.

I would say throwing JS errors should be used for showing some issues with the JS code itself.